controllers/numaresourcesoperator_controller.go

```diff
@@ -295,6 +297,30 @@ func (r *NUMAResourcesOperatorReconciler) syncDaemonSetsStatuses(ctx context.Con
 	return dsStatuses, true, nil
 }
 
+func syncNodeGroupsStatusPerMCPWithOperatorStatus(instance *nropv1.NUMAResourcesOperator) []nropv1.NodeGroupStatus {
+	ngStatuses := []nropv1.NodeGroupStatus{}
+	for _, mcp := range instance.Status.MachineConfigPools {
+		cfg := mcp.Config.DeepCopy()
+		ngStatuses = append(ngStatuses, nropv1.NodeGroupStatus{PoolName: mcp.Name, Config: cfg})
+	}
+	return ngStatuses
+}
+
+func syncNodeGroupsStatusesDaemonSets(instance *nropv1.NUMAResourcesOperator) []nropv1.NodeGroupStatus {
+	for _, ds := range instance.Status.DaemonSets {
+		copyOfDS := ds.DeepCopy()
+		extractedPoolName := objectnames.ExtractPoolNameFromRTEDaemonset(copyOfDS.Name, instance.Name)
+
+		for idx, ng := range instance.Status.NodeGroups {
+			if ng.PoolName == extractedPoolName {
+				instance.Status.NodeGroups[idx].DaemonSet = copyOfDS
+				break
+			}
+		}
+	}
+	return instance.Status.NodeGroups
+}
+
```

Comment:
this looks not wrong but we can probably set it straight away instead of searching it back.
But it's an optimization/cleanup we can do later.

Reply:
there's another way which includes changes to `reconcileResourcesDaemonSet` and which I'd like. We can generate quite easily the mapping between nodegroups and daemonsets while we build them. We can start with something like
```
diff --git a/controllers/numaresourcesoperator_controller.go b/controllers/numaresourcesoperator_controller.go
index 6e2be478..46e16937 100644
--- a/controllers/numaresourcesoperator_controller.go
+++ b/controllers/numaresourcesoperator_controller.go
@@ -462,8 +462,21 @@ func (r *NUMAResourcesOperatorReconciler) syncNUMAResourcesOperatorResources(ctx
                rteupdate.DaemonSetHashAnnotation(r.RTEManifests.DaemonSet, cmHash)
        }
 
+       dssByMCP := make(map[string]nropv1.NamespacedName)
+       processor := func(mcpName string, gdm *rtestate.GeneratedDesiredManifest) error {
+               err := daemonsetUpdater(mcpName, gdm)
+               if err != nil {
+                       return err
+               }
+               dssByMCP[mcpName] = nropv1.NamespacedName{
+                       Namespace: gdm.DaemonSet.GetNamespace(),
+                       Name:      gdm.DaemonSet.GetName(),
+               }
+               return nil
+       }
+
        existing := rtestate.FromClient(ctx, r.Client, r.Platform, r.RTEManifests, instance, trees, r.Namespace)
-       for _, objState := range existing.State(r.RTEManifests, daemonsetUpdater) {
+       for _, objState := range existing.State(r.RTEManifests, processor) {
                if objState.Error != nil {
                        // We are likely in the bootstrap scenario. In this case, which is expected once, everything is fine.
                        // If it happens past bootstrap, still carry on. We know what to do, and we do want to enforce the desired state.
```
with a mapping like` dssByMCP` (trivial to extend to `dssByNodeGroups`, you name it, this code should become simpler

This can be cleaned up later, though

Reply:
thanks for raising, addressed!